packages/wrangler/src/images/local.ts

```diff
@@ -0,0 +1,187 @@
+import type { ImageInfoResponse } from "@cloudflare/workers-types/experimental";
+import type { Sharp } from "sharp";
+
+const Transform = z.object({
+	imageIndex: z.number().optional(),
+	rotate: z.number().optional(),
+	width: z.number().optional(),
+	height: z.number().optional(),
+});
+
+const Transforms = z.array(Transform);
+
+export async function imagesLocalFetcher(request: Request): Promise<Response> {
+	const data = await request.formData();
+
+	const body = data.get("image");
+	if (!body || !(body instanceof File)) {
+		return errorResponse(400, 9523, "ERROR: Expected image in request");
+	}
+
+	const transformer = sharp(await body.arrayBuffer(), {});
+
+	const url = new URL(request.url);
+
+	if (url.pathname == "/info") {
+		return runInfo(transformer);
+	} else {
+		const badTransformsResponse = errorResponse(
+			400,
+			9523,
+			"ERROR: Expected JSON array of valid transforms in transforms field"
+		);
+		try {
+			const transformsJson = data.get("transforms");
+
+			if (typeof transformsJson !== "string") {
+				return badTransformsResponse;
+			}
+
+			const transforms = Transforms.safeParse(JSON.parse(transformsJson));
+
+			if (!transforms.success) {
+				return badTransformsResponse;
+			}
+
+			const outputFormat = data.get("output_format");
+
+			if (outputFormat != null && typeof outputFormat !== "string") {
+				return errorResponse(
+					400,
+					9523,
+					"ERROR: Expected output format to be a string if provided"
+				);
+			}
+
+			return runTransform(transformer, transforms.data, outputFormat);
+		} catch (e) {
+			return badTransformsResponse;
+		}
+	}
+}
+
+async function runInfo(transformer: Sharp): Promise<Response> {
+	const metadata = await transformer.metadata();
+
+	let mime: string | null = null;
+	switch (metadata.format) {
+		case "jpeg":
+			mime = "image/jpeg";
+			break;
+		case "svg":
+			mime = "image/svg+xml";
+			break;
+		case "png":
+			mime = "image/png";
+			break;
+		case "webp":
+			mime = "image/webp";
+			break;
+		case "gif":
+			mime = "image/gif";
+			break;
+		case "avif":
+			mime = "image/avif";
+			break;
+		default:
+			return errorResponse(415, 9520, "ERROR: Unsupported image type");
+	}
+
+	let resp: ImageInfoResponse;
+	if (mime == "image/svg+xml") {
+		resp = {
+			format: mime,
+		};
+	} else {
+		if (!metadata.size || !metadata.width || !metadata.height) {
+			return errorResponse(
+				500,
+				9523,
+				"ERROR: Expected size, width and height for bitmap input"
+			);
+		}
+
+		resp = {
+			format: mime,
+			fileSize: metadata.size,
+			width: metadata.width,
+			height: metadata.height,
+		};
+	}
+
+	return Response.json(resp);
+}
+
+async function runTransform(
+	transformer: Sharp,
+	transforms: z.infer<typeof Transforms>,
+	outputFormat: string | null
+): Promise<Response> {
+	for (const transform of transforms) {
+		if (transform.imageIndex !== undefined && transform.imageIndex !== 0) {
+			// We don't support draws, and this transform doesn't apply to the root
```

Comment:
I have no context here, but I was wondering if this is smth we should surface to users?